create_framework/01-introduction.rst

```diff
@@ -0,0 +1,139 @@
+------------------------------------------------
+
+Why would you like to create your own framework in the first place? If you
+look around, everybody will tell you that it's a bad thing to reinvent the
+wheel and that you'd better choose an existing framework and forget about
+creating your own altogether. Most of the time, they are right but there are
+a few good reasons to start creating your own framework:
+
+* To learn more about the low level architecture of modern web frameworks in
+  general and about the Symfony2 full-stack framework internals in particular;
+
+* To create a framework tailored to your very specific needs (just be sure
+  first that your needs are really specific);
+
+* To experiment creating a framework for fun (in a learn-and-throw-away
+  approach);
+
+* To refactor an old/existing application that needs a good dose of recent web
+  development best practices;
+
+* To prove the world that you can actually create a framework on your own (...
+  but with little effort).
+
+This tutorial will gently guide you through the creation of a web framework,
+one step at a time. At each step, you will have a fully-working framework that
+you can use as is or as a start for your very own. We will start with simple
+frameworks and more features will be added with time. Eventually, you will have
+a fully-featured full-stack web framework.
+
+And of course, each step will be the occasion to learn more about some of the
+Symfony2 Components.
+
+.. tip::
+
+    If you don't have time to read the whole book, or if you want to get
+    started fast, you can also have a look at `Silex`_, a micro-framework
+    based on the Symfony2 Components. The code is rather slim and it leverages
+    many aspects of the Symfony2 Components.
+
+Many modern web frameworks advertize themselves as being MVC frameworks. We
+won't talk about the MVC pattern as the Symfony2 Components are able to create
+any type of frameworks, not just the ones that follow the MVC architecture.
+Anyway, if you have a look at the MVC semantics, this book is about how to
+create the Controller part of a framework. For the Model and the View, it
+really depends on your personal taste and you can use any existing
+third-party libraries (Doctrine, Propel, or plain-old PDO for the Model; PHP
+or Twig for the View).
+
+When creating a framework, following the MVC pattern is not the right goal. The
+main goal should be the **Separation of Concerns**; this is probably the only
+design pattern that you should really care about. The fundamental principles of
+the Symfony2 Components are focused on the HTTP specification. As such, the
+frameworks that we are going to create should be more accurately labelled as
+HTTP frameworks or Request/Response frameworks.
+
+Before we start
+---------------
+
+Reading about how to create a framework is not enough. You will have to follow
+along and actually type all the examples we will work on. For that, you need a
```

Comment:
"[...] we will work on" -> "[...] you will see." or something else as long as it avoids the first person usage.